ts-packages/transform/src/preprocess/handlebars.ts

```diff
@@ -34,19 +34,44 @@ Handlebars.registerHelper("generateActionMember", function (fn: ParameterNodeDef
     let code: string[] = [];
     let plainType = fn.type.plainTypeNode;
     if (plainType == 'string') {
-        code.push(`public ${fn.name}: string = "",`);
+        code.push(` ${fn.name}: string = "";`);
     } else if (plainType == 'string[]') {
-        code.push(`public ${fn.name}: string[] = [],`);
+        code.push(` ${fn.name}: string[] = [];`);
     } else {
         if (TypeHelper.isPrimitiveType(fn.type.typeKind)) {
-            code.push(`public ${fn.name}: ${plainType},`);
+            code.push(` ${fn.name}: ${plainType};`);
         } else {
-            code.push(`public ${fn.name}!: ${plainType},`);
+            code.push(` ${fn.name}!: ${plainType};`);
         }
     }
     return code.join(EOL);
 });
 
+Handlebars.registerHelper("generateActionParam", function (fn: ParameterNodeDef) {
+    let code: string[] = [];
+    let plainType = fn.type.plainTypeNode;
+
+    if (plainType == 'string') {
+        code.push(` ${fn.name}: string = "",`);
+    } else if (plainType == 'string[]') {
+        code.push(` ${fn.name}: string[] = [],`);
+    } else {
+        if (TypeHelper.isPrimitiveType(fn.type.typeKind)) {
```

Comment:
primitive type may be a string, assigning 0 to a string will cause an error.

Reply:
@learnforpractice doesnt the plainType check above for == 'string' cover that?

Reply:
Yes, ignore my previous comment. But I still smell that we need a refactor of this function. I will approve this PR, so we can iterate fast.